src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -351,6 +338,73 @@ private SmugglerProgressBase.Counts ProcessIndexes(SmugglerResult result)
+                }
+
+                actions.WriteIndex(indexDefinition);
+            }
+            catch (Exception e)
+            {
+                var exceptionMessage = e.Message;
+                if (exceptionMessage.Contains("No overload for method 'LoadDocument' takes 1 arguments"))
+                {
+                    exceptionMessage = $"{e.Message}" + Environment.NewLine +
+                            "LoadDocument requires a generic argument for its usage (the type of the loaded document)" + Environment.NewLine +
+                            "For example: " + Environment.NewLine +
+                                "\tfrom doc in doc.Orders" + Environment.NewLine +
+                                "\tlet company = LoadDocument(doc.Company, \"Companies\")" + Environment.NewLine +
+                                "\tselect new {" + Environment.NewLine +
+                                    "\t\tCompanyName: comapny.Name" + Environment.NewLine +
+                                "\t}" + Environment.NewLine;
+                }
+                else if (retryOnError && 
+                         (exceptionMessage.Contains("The name 'AbstractIndexCreationTask' does not exist in the current context") ||
+                          exceptionMessage.Contains("The name 'SpatialIndex' does not exist in the current context") ||
+                          exceptionMessage.Contains("The type or namespace name 'Abstractions' does not exist in the namespace 'Raven' (are you missing an assembly reference?)")))
+                {
+                    var maps = new HashSet<string>();
+                    foreach (var map in indexDefinition.Maps)
+                    {
+                        maps.Add(ReplaceLegacyMethods(map));
+                    }
+
+                    if (indexDefinition.Reduce != null)
+                    {
+                        indexDefinition.Reduce = ReplaceLegacyMethods(indexDefinition.Reduce);
+                    }
+
+                    indexDefinition.Maps = maps;
+                    WriteIndex(result, indexDefinition, actions, retryOnError: false);
+                    return;
+                }
+
+                result.Indexes.ErroredCount++;
+                var errorMessage = $"Could not write index '{indexDefinition.Name}', error: {exceptionMessage}" + Environment.NewLine +
+                                   $"Maps: [{Environment.NewLine}{string.Join($", {Environment.NewLine}", indexDefinition.Maps)}{Environment.NewLine}]";
+
+                if (indexDefinition.Reduce != null)
+                {
+                    errorMessage += Environment.NewLine + $"Reduce: {indexDefinition.Reduce}";
+                }
+                result.AddError(errorMessage);
+            }
+
+            string ReplaceLegacyMethods(string str)
+            {
+                return str
+                    .Replace("AbstractIndexCreationTask.SpatialGenerate", "CreateSpatialField")
```

Comment:
Let's not do that, this will not work as expected, the arguments do not match.

Reply:
for those that don't match we display an error:
(22,74): error CS1501: No overload for method 'CreateSpatialField' takes 4 arguments